storage/Devices/Bcache.h

```diff
@@ -33,80 +36,122 @@ namespace storage
     class BcacheCset;
 
 
+    /**
+     * Bcache types
+     */
+    enum class BcacheType {
+	BACKED, FLASH_ONLY
+    };
+
+
     /**
      * The Cache mode attribute.
      */
     enum class CacheMode {
 	WRITETHROUGH, WRITEBACK, WRITEAROUND, NONE
     };
 
-
     /**
-     * A <a href="https://www.kernel.org/doc/Documentation/bcache.txt">bcache</a>
-     * device.
+     * A <a href="https://www.kernel.org/doc/Documentation/bcache.txt">Bcache</a> device.
+     *
+     * Bcache technology supports two kinds of bcache devices. Bcache devices with a backing
+     * device and also Bcache devices directly created over a caching set (without backing device
+     * associated to it). This second type is known as Flash-only Bcache.
      */
     class Bcache : public Partitionable
     {
     public:
 
 	static Bcache* create(Devicegraph* devicegraph, const std::string& dm_name);
+	static Bcache* create(Devicegraph* devicegraph, const std::string& dm_name, BcacheType type);
 	static Bcache* load(Devicegraph* devicegraph, const xmlNode* node);
 
+	BcacheType get_type() const;
+
 	unsigned int get_number() const;
 
 	/**
 	 * Get the BlkDevice used as backing device.
+	 *
+	 * This method does not make sense for Flash-only Bcache devices.
+	 *
+	 * @throw Exception
 	 */
-	const BlkDevice* get_blk_device() const;
+	const BlkDevice* get_backing_device() const;
 
 	/**
-	 * Returns true if a BcacheCset is attached.
+	 * @copydoc get_backing_device
+	 */
+	const BlkDevice* get_blk_device() const ST_DEPRECATED { return get_backing_device(); }
```

Comment:
I discussed this with Stefan and we are not sure whether implementing the function in the header does not break the ABI. E.g. the compiler could decide to not include the function in the library since it is inlined.

Reply:
Ok, makes sense. Moved to .cc file.